tutorials/models_mapping/plot_1_tensor_models.py

```diff
@@ -0,0 +1,333 @@
+           )
+mod[k_block] = m_block
+
+# Plot
+Fig = plt.figure(figsize=(5, 5))
+Ax = Fig.add_subplot(111)
+mesh.plotSlice(modMap*mod, normal='Y', ax=Ax, ind=int(mesh.nCy/2), grid=True)
+Ax.set_title('Model slice at y = -2.5 m')
+plt.show()
+
+
+#############################################
+# Combo Maps
+# ----------
+#
+# Here we demonstrate how combo maps can be used to create a single mapping
+# from the model to the mesh. In this case, our model consists of
+# log-conductivity values but we want to plot the resistivity. To accomplish
+# this we must take the exponent of our model values, then take the reciprocal,
+# then map from below surface cell to the mesh.
+#
+
+mesh = makeExampleMesh()
+
+m_back = np.log(1./100.)
+m_dyke = np.log(1./40.)
+m_block = np.log(1./70.)
+
+# Define surface topography
+[xx, yy] = np.meshgrid(mesh.vectorNx, mesh.vectorNy)
+zz = -3*np.exp((xx**2 + yy**2) / 75**2) + 40.
+topo = np.c_[Utils.mkvc(xx), Utils.mkvc(yy), Utils.mkvc(zz)]
+
+# Find cells below topography
+m_air = 0.
+actv = Utils.surface2ind_topo(mesh, topo, 'N')
+actvMap = Maps.InjectActiveCells(mesh, actv, m_air)
+
+# Define the model on subsurface cells
+mod = m_back*np.ones(actv.sum())
+k_dyke = (mesh.gridCC[actv, 0] > 20.) & (mesh.gridCC[actv, 0] < 40.)
+mod[k_dyke] = m_dyke
+k_block = ((mesh.gridCC[actv, 0] > -40.) & (mesh.gridCC[actv, 0] < -10.) &
+           (mesh.gridCC[actv, 1] > -30.) & (mesh.gridCC[actv, 1] < 30.) &
+           (mesh.gridCC[actv, 2] > -40.) & (mesh.gridCC[actv, 2] < 0.)
+           )
+mod[k_block] = m_block
+
+# Define a single mapping from model to mesh
+expMap = Maps.ExpMap()
+recMap = Maps.ReciprocalMap()
+modMap = Maps.ComboMap([actvMap, recMap, expMap])
+
+# Plot
+Fig = plt.figure(figsize=(5, 5))
+Ax = Fig.add_subplot(111)
+mesh.plotSlice(modMap*mod, normal='Y', ax=Ax, ind=int(mesh.nCy/2), grid=True)
+Ax.set_title('Model slice at y = -2.5 m')
+plt.show()
+
+
+#############################################
+# Models with arbitrary shapes
+# ----------------------------
+#
+# Here we show how model building utilities are used to make more complicated
+# structural models. The process of adding a new unit is twofold: 1) we must
+# find the indicies for mesh cells that lie within the new unit, 2) we
+# replace the prexisting physical property value for those cells.
+#
+
+mesh = makeExampleMesh()
+
+m_back = 100.
+m_dyke = 40.
+m_sph = 70.
+
+# Define surface topography
+[xx, yy] = np.meshgrid(mesh.vectorNx, mesh.vectorNy)
+zz = -3*np.exp((xx**2 + yy**2) / 75**2) + 40.
+topo = np.c_[Utils.mkvc(xx), Utils.mkvc(yy), Utils.mkvc(zz)]
+
+# Set active cells and define unit values
+m_air = 0.
+actv = Utils.surface2ind_topo(mesh, topo, 'N')
+modMap = Maps.InjectActiveCells(mesh, actv, m_air)
+
+# Define model for cells under the surface topography
+mod = m_back*np.ones(actv.sum())
+
+# Add a sphere
+sphInd = Utils.ModelBuilder.getIndicesSphere(np.r_[-25., 0., -15.],
```

Comment:
Rather than using the `ModelBuilder` code - lets just stick with mappings for now. I think the model builder stuff might eventually get folded into `maps` or moved elsewhere, so mappings are more important to highlight in current tutorials

Reply:
I disagree. I made a conscious decision to setup the tutorials this way with the current status of the SimPEG package. I wanted the users to correctly differentiate the model from the physical property distribution on a mesh. And I wanted to clearly define that a mapping could be defined that would go from the model space to the mesh. Feedback from @sarahgarre  suggests that my tutorial was successful in doing this.

If the way in which functionality like 'ModelBuilder.getIndicesSphere' is called changes in the future, we can always change it in a patch.

Reply:
Hi Dev, I appreciate that this is the current status. I am trying to advise on how we want to be encouraging users to use SimPEG to avoid frustration as we continue to improve the code. 

In this case, the model builder code in its current form is likely to change, whereas the maps are not. So we want to encourage use of maps. (via the `Maps` module rather that the `Utils.ModelBuilder`). 

If you feel strongly that a `Utils.ModelBuilder` example tutorial should exist, then would it be okay with you if we moved it to a separate tutorial? That way we would have one which uses only maps, and one which uses the model builder code. Separating the two will hopefully clarify that `Maps` and `Utils.ModelBuilder` are not completely interchangeable.